internal/types/path.go

```diff
@@ -181,3 +183,112 @@ func removeByPath(v any, path Path) {
 		// no such path: scalar value
 	}
 }
+
+// insertByPath inserts missing parts of the path into Document.
+func insertByPath(doc *Document, path Path) error {
+	var next any = doc
+
+	var insertedPath Path
+	for _, pathElem := range path.TrimSuffix().Slice() {
+		insertedPath = insertedPath.Append(pathElem)
+
+		v, err := doc.GetByPath(insertedPath)
+		if err != nil {
+			suffix := len(insertedPath.Slice()) - 1
+			if suffix < 0 {
+				panic("invalid path")
+			}
+
+			switch v := next.(type) {
+			case *Document:
+				must.NoError(v.Set(insertedPath.Slice()[suffix], must.NotFail(NewDocument())))
+			case *Array:
+				_, err := strconv.Atoi(insertedPath.Slice()[suffix])
+				if err != nil {
+					return fmt.Errorf(
+						"Cannot create field '%s' in element {%s: %s}",
+						pathElem,
+						insertedPath.Slice()[suffix-1],
+						formatDocumentValue(v),
+					)
+				}
+			}
+
+			next = must.NotFail(doc.GetByPath(insertedPath)).(*Document)
+
+			continue
+		}
+
+		next = v
+	}
+
+	return nil
+}
+
+// formatDocumentValue formats document value for error message output.
+func formatDocumentValue(v any) string {
```

Comment:
Minor note. Based on what this function does, I'd call it `formatAnyValue`, not just `Document` :)  I guess, the idea is something like `Format Document's Value` but I'm not sure how to name such a function.

Reply:
`formatAnyValue` sounds reasonable!